File: Dev/unitTest_Engine_cs/Graphics/2D/TextureObject2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._2D
{
    class TextureObject2D : EngineTest
    {
        public TextureObject2D()
            : base(20)
        {
        }

        protected override void OnStart()
        {
            var scene = new asd.Scene();
            var layer = new asd.Layer2D();
            var texture = asd.Engine.Graphics.CreateTexture2D("Data/Texture/Sample1.png");
            var textureObj1 = new asd.TextureObject2D();
            var textureObj2 = new asd.TextureObject2D();

            layer.AddObject(textureObj1);
            layer.AddObject(textureObj2);
            scene.AddLayer(layer);
            asd.Engine.ChangeScene(scene);

            textureObj1.Texture = texture;
            textureObj1.Src = new asd.RectF(256, 256, 256, 256);
            textureObj1.Position=new asd.Vector2DF(320, 240);

            textureObj2.Texture = texture;
            textureObj2.Src = new asd.RectF(0, 0, 256, 256);
            textureObj2.Position = new asd.Vector2DF(0, 0);
            textureObj2.TurnLR = true;
        }

        protected override void OnUpdating()
        {

        }
    }
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/TextureObject2D.cpp

#include <gtest/gtest.h>
#include <Altseed.h>
#include "../../EngineTest.h"

using namespace std;
using namespace asd;

class Graphics_TextureObject2D : public EngineTest
{
public:
	Graphics_TextureObject2D(bool isOpenGLMode)
		: EngineTest(asd::ToAString("TextureObject2D"), isOpenGLMode, 20)
	{
	}

private:

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();
		auto texture = asd::Engine::GetGraphics()->CreateTexture2D(asd::ToAString("Data/Texture/Sample1.png").c_str());
		auto textureObj1 = make_shared<TextureObject2D>();
		auto textureObj2 = make_shared<TextureObject2D>();

		layer->AddObject(textureObj1);
		layer->AddObject(textureObj2);
		scene->AddLayer(layer);
		asd::Engine::ChangeScene(scene);

		textureObj1->SetTexture(texture);
		textureObj1->SetSrc(asd::RectF(256, 256, 256, 256));
		textureObj1->SetPosition(asd::Vector2DF(320, 240));

		textureObj2->SetTexture(texture);
		textureObj2->SetSrc(asd::RectF(0, 0, 256, 256));
		textureObj2->SetPosition(asd::Vector2DF(400, 320));
		textureObj2->SetTurnLR(true);
	}

	void OnUpdating()
	{
	}
};


ENGINE_TEST(Graphics, TextureObject2D)
